odoo/models.py

```diff
@@ -4863,7 +4863,8 @@ def copy(self, default=None):
         self.ensure_one()
         vals = self.with_context(active_test=False).copy_data(default)[0]
         # To avoid to create a translation in the lang of the user, copy_translation will do it
-        new = self.with_context(lang=None).create(vals)
+        original_lang = self.env.context.get('lang')
+        new = self.with_context(lang=None).create(vals).with_context(lang=original_lang)
```

Comment:
This potentially creates yet another environment if `self.env.context` does not contain key `lang`.
I suggest to use `new.with_env(self.env)` to get the record with the original environment.
It's way simpler and more efficient.